content/influxdb/v2.1/api-guide/guides-outline.md

```diff
@@ -0,0 +1,663 @@
+const url =
+  write_endpoint && write_endpoint !== '/mqtt' ? write_endpoint : '/influx'
+const influxDB = new InfluxDB({url, token})
+const writeApi = influxDB.getWriteApi(org, bucket, 'ns', {
+  batchSize: batchSize + 1,
+  defaultTags: {clientId: id},
+})
+try {
+  // write random temperatures
+  const point = new Point('environment') // reuse the same point to spare memory
+  onProgress(0, 0, totalPoints)
+
+  const writePoint = async (time: number) => {
+    const gpx = getGPX(time)
+    point
+      .floatField('Temperature', generateTemperature(time))
+      .floatField('Humidity', generateHumidity(time))
+      .floatField('Pressure', generatePressure(time))
+      .intField('CO2', generateCO2(time))
+      .intField('TVOC', generateTVOC(time))
+      .floatField('Lat', gpx[0] || state.config.default_lat || 50.0873254)
+      .floatField('Lon', gpx[1] || state.config.default_lon || 14.4071543)
+      .tag('TemperatureSensor', 'virtual_TemperatureSensor')
+      .tag('HumiditySensor', 'virtual_HumiditySensor')
+      .tag('PressureSensor', 'virtual_PressureSensor')
+      .tag('CO2Sensor', 'virtual_CO2Sensor')
+      .tag('TVOCSensor', 'virtual_TVOCSensor')
+      .tag('GPSSensor', 'virtual_GPSSensor')
+      .timestamp(String(time) + '000000')
+    writeApi.writePoint(point)
+
+    pointsWritten++
+    if (pointsWritten % batchSize === 0) {
+      await writeApi.flush()
+      onProgress(
+        (pointsWritten / totalPoints) * 100,
+        pointsWritten,
+        totalPoints
+      )
+    }
+  }
+
+  if (missingDataTimeStamps && missingDataTimeStamps.length)
+    for (const timestamp of missingDataTimeStamps)
+      await writePoint(timestamp)
+  else
+    while (lastTime < toTime) {
+      lastTime += 60_000 // emulate next minute
+      await writePoint(lastTime)
+    }
+  await writeApi.flush()
+} finally {
+  await writeApi.close()
+}
+onProgress(100, pointsWritten, totalPoints)
+}
+```
+
+## Query InfluxDB
+
+See [Query with the API](/influxdb/v2.1/query-data/execute-queries/influx-api/)
+
+## Query data for visualizations
+
+[@influxdata/giraffe](https://github.com/influxdata/giraffe), the React-based
+visualization library for InfluxDB v2, provides an extensive range of data visualization components and support for Flux query results.
+
+Flux queries return data in CSV format.
+When generating the table through a Flux result:
+
+call the `fromFlux()` utility function on the CSV generated by Flux
+get the table in the returned object from calling fromFlux
+
+{{% note %}}
+Use '@influxdata/giraffe' to render gauges, plots, tables, and other visualizations in your own dashboard UI.
+To learn more about Giraffe visualizations, options, and the columnar interface see the [Giraffe Quick Start](https://github.com/influxdata/giraffe/blob/master/giraffe/README.md#quick-start).
+{{% /note %}}
+
+### Query and transform data with the Javascript client library
+
+Although you can use the `fromFlux()` '@influxdata/giraffe' utility directly,
```

Comment:
I think we can move the description of `fromFlux()` and push it after the code snippet. The code snippet is guiding us on how to use `queryToTable` and a description of `fromFlux()` doesn't seem to be necessary beforehand